_posts/en/posts/2016-01-25-segwit-benefits.md

```diff
@@ -0,0 +1,321 @@
+
+## Compact fraud proofs
+
+As the Bitcoin userbase expands, validating the entire blockchain
+naturally becomes more expensive. To maintain the decentralised, trustless
+nature of Bitcoin, it is important to allow those who cannot afford to
+validate the entire blockchain to at least be able to cheaply validate
+as much of it as they can afford.
+
+Segwit improves the situation here by allowing a future soft-fork to
+extend the witness structure to include commitment data, which will allow lightweight (SPV)
+clients to enforce consensus rules such such as the number of bitcoins
+introduced in a block, the size of a block, and the number of sigops
+used in a block.
+
+### Who benefits?
+
+Fraud proofs allow SPV users to help enforce Bitcoin's consensus rules,
+which will potentially greatly increase the security of the Bitcoin
+network as a whole, as well as reduce the ways in which individual users
+can be attacked.
+
+These fraud proofs can be added to the witness data structure as part
+of a future soft-fork, and they'll help SPV clients enforce the rules
+even on transactions that don't make use of the segwit features.
+
+## Signature pruning
+
+For old transactions, signatures may be less interesting -- for example,
+some SPV clients simply don't check signatures, trusting that miners
+have already done that.  Even Bitcoin Core does not check signatures for
+transactions prior to the most recent checkpoint. At present, however,
+signature data is an integral part of the transaction and must be
+present in order to calculate the transaction hash.
+
+Segregating the signature data allows nodes that aren't interested in
+signature data to prune it from the disk, or never download it in the
+first place, saving resources.
+
+### Who benefits?
+
+As more transactions use segwit addresses, people running pruned or SPV
+nodes will be able to operate with less bandwidth and disk space.
+
+## Moving towards a single combined block limit
+
+Currently there are two consensus-enforced limits on blocksize: the
+block can be no larger than 1MB and, independently, there can be no
+more than 20,000 signature checks performed across the transactions in
+the block.
+
+Finding the most profitable set of transactions to include
+in a block given a single limit is an instance of the knapsack
+problem, which can be easily solved almost perfectly with a simple
+greedy algorithm. However adding the second constraint makes finding a
+good solution very hard in some cases, and this theoretical problem has
+been exploited in practice to force blocks to be mined at a size well
+below capacity.
+
+It is not possible to solve this problem without either a hardfork,
+or substantially decreasing the block size.  Since segwit can't fix the
+problem, it settles on not making it worse: in particular, segwit's newly introduced limit
+on the new concept of witness data is implemented as a weighted sum
+of the existing block size and the new witness data.
```

Comment:
The witness data is not new, it is merely moved.

"in particular, segwit's newly introduced size limit is implemented with a weight on the particular kinds of data within the block."